### PR TITLE
Get lessons by standard to generate progress boxes

### PR DESCRIPTION
**Pull request by Erin007**
[LP-1127](https://codedotorg.atlassian.net/browse/LP-1127)
[spec](https://docs.google.com/document/d/1xlm_syljGXIapl72GaKxN63V7kt6syHyvWHuqZhTCfE/edit#heading=h.ax62vxl69vcw)

On the Standards View of the Progress tab for each CSTA standard associated with a CSF script we now show a Progress Box for each stage/lesson tagged with that standard based on real data! 

Screen Shot 2020-01-31 at 9 40 50 PM

Tooltips for each box show the lesson name and how many students are in the currently selected section. 
Screen Shot 2020-01-31 at 9 43 38 PM

Up next is determining whether the lesson is completed (ie. whether the Progress Box should be green or not) and how many students in the section have completed the lesson (for the tooltip, currently "undefined").

### CODE REVIEW

File: apps/src/templates/sectionProgress/standards/StandardsProgressTable.jsx [hunk 1]
```diff
@@ -151,14 +151,12 @@ class StandardsProgressTable extends Component {
 
   getNumCompleted = (standard, index) => {
```
Comment (clareconstantine) [inline apps/src/templates/sectionProgress/standards/StandardsProgressTable.jsx]: could we add a comment to this method to clarify that it's the number of completed lessons for the standard (as opposed to something like the number of students who have completed the standard)?

File: apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js [hunk 2]
```diff
@@ -33,6 +33,42 @@ export function getLessonsCompletedByStandardForScript(script) {
...
+
+    const stages =
+      state.sectionProgress.scriptDataByScript[state.scriptSelection.scriptId]
+        .stages;
+
+    const numStudents =
+      state.teacherSections.sections[state.teacherSections.selectedSectionId]
+        .studentCount;
+
+    standards.forEach(standard => {
```
Comment (clareconstantine) [inline apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js:54]: this logic feels well-organized and easy to follow :)

File: dashboard/app/models/script.rb [hunk 3]
```diff
@@ -678,7 +678,15 @@ def has_standards_associations?
   end
 
   def standards
-    stages.map(&:standards).flatten.uniq.map(&:summarize)
+    standards = stages.map(&:standards).flatten.uniq
+    standards_with_stages = []
+    standards.each do |standard|
+      standards_summary = standard.summarize
+      stages_by_standard = stages && standard.stages
```
Comment (clareconstantine) [inline dashboard/app/models/script.rb]: It looks like the intersection operator is a single ampersand: https://ruby-doc.org/core-2.4.1/Array.html#method-i-26

```suggestion
      stages_by_standard = stages & standard.stages
```

File: dashboard/app/models/script.rb [hunk 4]
```diff
@@ -678,7 +678,15 @@ def has_standards_associations?
   end
 
   def standards
-    stages.map(&:standards).flatten.uniq.map(&:summarize)
+    standards = stages.map(&:standards).flatten.uniq
+    standards_with_stages = []
+    standards.each do |standard|
+      standards_summary = standard.summarize
```
Comment (clareconstantine) [inline dashboard/app/models/script.rb]: nit - I would call it `standard_summary` since this variable only holds the summary of one standard at a time

**Review by clareconstantine — COMMENTED** [review]
a couple small things, but this is looking good!

File: apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js (hunk 2 above)
Comment (dmcavoy) [reply apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js:54]: +1!

**Review by dmcavoy — APPROVED** [review]
Awesome work Erin! This has come together so nicely!

**Pull request merged**